docs/practices/unit_testing.rst

```diff
@@ -29,7 +29,18 @@ expected.
 
 It's also worth noting that if you want to write exploratory tests as you develop
 your code, but you *do not* want those tests to be included in automated test runs,
-feel free to place those tests in a directory outside of the ``./tests`` directory.
+feel free to place those tests in a directory outside of the ``./tests`` and 
+``./src`` directories.
 
 Note that ``pytest`` will recursively search subdirectories inside of ``./tests``
 while searching for tests to run.
+
+doctests
+-------------------------------------------------------------------------------
+
+In addition to the usual ways of writing unit tests with pytest, our template
+supports tests embedded in documentation using pytest's 
+`doctest <https://doc.pytest.org/en/latest/how-to/doctest.html>`_ component. 
+Documentation comments in all source files, as well as ``.rst`` files in the ``./docs`` 
+directory can contain doctests in the format outlined 
+`here <https://doc.pytest.org/en/latest/how-to/doctest.html>`_.
```

Comment:
Should it be some other link?

Reply:
I'm not sure there's a better one. This change doesn't include anything `.rst` specific, so you just have to use the normal syntax in your `.rst` files.

AFAICT that one page is the only obviously authoritative place on the internet that explains the syntax. If we somehow do something special and more `.rst` friendly we could amend this.

Reply:
It is not, docstring is PEP257
https://peps.python.org/pep-0257/

Reply:
Doctest is a part of PEP287, and it is support by the standard library's unitest module